contracts/dao/nopeDAO/NopeDAO.sol

```diff
@@ -0,0 +1,172 @@
+import {ERC20VotesComp} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20VotesComp.sol";
+import {CoreRef} from "../../refs/CoreRef.sol";
+import {TribeRoles} from "../../core/TribeRoles.sol";
+import {GovernorQuickReaction} from "./GovernorQuickReaction.sol";
+
+contract NopeDAO is
+    Governor,
+    GovernorSettings,
+    GovernorVotesComp,
+    GovernorQuickReaction,
+    CoreRef
+{
+    /// @notice Initial quorum required for a Nope proposal
+    uint256 private _quorum = 10_000_000e18;
+
+    /// @notice Additional governance events
+    event QuorumUpdated(uint256 oldQuorum, uint256 newQuorum);
+
+    constructor(ERC20VotesComp _tribe, address _core)
+        Governor("NopeDAO")
+        GovernorSettings(
+            1, /* 1 block */
+            345600, /* 4 days */
+            0
+        )
+        GovernorVotesComp(_tribe)
+        GovernorQuickReaction()
+        CoreRef(_core)
+    {}
+
+    function quorum(uint256 blockNumber)
+        public
+        view
+        override
+        returns (uint256)
+    {
+        return _quorum;
+    }
+
+    ////////////     GOVERNOR ONLY FUNCTIONS     //////////////
+
+    /// @notice Override of a Governor Settings function, to restrict to Tribe GOVERNOR
+    function setVotingDelay(uint256 newVotingDelay)
+        public
+        override
+        onlyTribeRole(TribeRoles.GOVERNOR)
+    {
+        _setVotingDelay(newVotingDelay);
+    }
+
+    /// @notice Override of a Governor Settings function, to restrict to Tribe GOVERNOR
+    function setVotingPeriod(uint256 newVotingPeriod)
+        public
+        override
+        onlyTribeRole(TribeRoles.GOVERNOR)
+    {
+        _setVotingPeriod(newVotingPeriod);
+    }
+
+    /// @notice Override of a Governor Settings function, to restrict to Tribe GOVERNOR
+    function setProposalThreshold(uint256 newProposalThreshold)
+        public
+        override
+        onlyTribeRole(TribeRoles.GOVERNOR)
+    {
+        _setProposalThreshold(newProposalThreshold);
+    }
+
+    /// @notice Adjust quorum of NopeDAO. Restricted to GOVERNOR, not part of GovernorSettings
+    function setQuorum(uint256 newQuroum)
+        public
+        onlyTribeRole(TribeRoles.GOVERNOR)
+    {
+        uint256 oldQuorum = _quorum;
+        _quorum = newQuroum;
+        emit QuorumUpdated(oldQuorum, newQuroum);
+    }
+
+    // The following functions are overrides required by Solidity.
+
+    function votingDelay()
+        public
+        view
+        override(IGovernor, GovernorSettings)
+        returns (uint256)
+    {
+        return super.votingDelay();
+    }
+
+    function votingPeriod()
+        public
+        view
+        override(IGovernor, GovernorSettings)
+        returns (uint256)
+    {
+        return super.votingPeriod();
+    }
+
+    function getVotes(address account, uint256 blockNumber)
+        public
+        view
+        override(IGovernor, GovernorVotesComp)
```

Comment:
Why override, couldn't you just inherit?

Reply:
It's a Solidity requirement. This contract inherits from two base classes that define the same function with the same arguments - `getVotes()` is on the interface of `IGovernor` and implemented in `GovernorVotesComp()`. 

Solidity requires in that case that you use this pattern and override. This error trace is what you get when you just try to inherit instead:

![image](https://user-images.githubusercontent.com/20632187/161080275-83f2a213-0a56-4037-a658-e65477478ee9.png)